src/authorities/OverdueInstanceAuthority.ts

```diff
@@ -0,0 +1,53 @@
+import {inject, injectable} from 'inversify';
+import PS2AlertsInstanceInterface from '../interfaces/PS2AlertsInstanceInterface';
+import {getLogger} from '../logger';
+import {TYPES} from '../constants/types';
+import InstanceHandlerInterface from '../interfaces/InstanceHandlerInterface';
+
+@injectable()
+export default class OverdueInstanceAuthority {
+    private static readonly logger = getLogger('OverdueInstanceAuthority');
+
+    private readonly instanceHandler: InstanceHandlerInterface;
+
+    private timer: NodeJS.Timeout | null = null;
+
+    constructor(@inject(TYPES.instanceHandlerInterface) instanceHandler: InstanceHandlerInterface) {
+        this.instanceHandler = instanceHandler;
+    }
+
+    public run(): void {
+        OverdueInstanceAuthority.logger.debug('Created OverdueInstanceAuthority timer');
+
+        if (this.timer) {
+            OverdueInstanceAuthority.logger.error('Attempted to run OverdueInstanceAuthority timer when already defined!');
+            clearInterval(this.timer);
+        }
+
+        this.timer = setInterval(() => {
+            OverdueInstanceAuthority.logger.debug('Running OverdueInstanceAuthority overdue alert check');
+
+            const instances: PS2AlertsInstanceInterface[] = this.instanceHandler.getAllInstances();
+
+            instances.forEach((instance: PS2AlertsInstanceInterface) => {
+                if (instance.overdue()) {
```

Comment:
Just use filter on the instances to get the overdue. Any functionality that node provides us is always better then anything we will be able to write.